conductor/lib/maya_utils.py

```diff
@@ -1038,13 +992,243 @@ def _scrape_ass_file(ass_filepath, node_attrs):
+            attr = part.lstrip("-")
+        else:
+            args[attr].append(part)
+
+    command_args = {}
+    for flag, value in args.iteritems():
+        if len(value) is 1:
+            value = value[0]
+        command_args[flag] = value
+    return command_args
+
+
+def _is_number(string):
+    '''
+    Return True if the string can be cast to a digit/float, otherwise return False
+    '''
+    try:
+        float(string)
+        return True
+    except ValueError:
+        return False
+
+
+def scrape_palette_node(palette_node, palette_filepath):
+    '''
+    Inspect/scrape the xgen palette node and parse it's associated .xgen file for file dependencies
+    named. 
+    '''
+    file_paths = []
+    match = re.match("(.+)\.xgen", palette_filepath)
+    if match:
+        abc_patch_file = cmds.file("%s.abc" % (match.group(1)), expandName=True, query=True, withoutCopyNumber=True)
+        if os.path.isfile(abc_patch_file):
+            file_paths.append(abc_patch_file)
+
+    palette_dependencies = scrape_xgen_file(palette_filepath)
+    return file_paths + palette_dependencies
+
+
+def scrape_xgen_file(filepath):
+    '''
+    Scrape the given .xgen file for file dependencies.
+    '''
+    try:
+        palette_modules = parse_xgen_file(filepath)
+    except Exception:
+        logger.exception("Failed to parse XGen Palette file: %s", filepath)
+        return []
+
+    resources = common.load_resources_file()
+    xgen_attrs = resources.get("xgen_dependency_attrs") or {}
+
+    paths = []
+
+    for module_type, module_attrs in xgen_attrs.iteritems():
+        for module in palette_modules.get(module_type, []):
+            for module_attr_info in module_attrs:
+                for module_attr, attr_conditions in module_attr_info.iteritems():
+                    if _are_conditions_met(module, attr_conditions or {}):
+                        value = module.get(module_attr, "")
+                        # Convert the raw parsed string value to a list of values (split by whitespace),
+                        # filtering out any empty values.
+                        values = filter(None, [v.strip(' \t\r\n') for v in value.split()])
+                        if values:
+                            # Use the last item in the values. This is a huge assumption (that we'll always
+                            # only want the last item, but it's the best we have without a proper
+                            # xgen api
+                            paths.append(values[-1])
+    return paths
+
+
+def _are_conditions_met(module, attr_conditions):
+    '''
+    Return True if the all of the given conditions are met for the the given xgen module values
+    '''
+    for condition_attr, condition_value in attr_conditions.iteritems():
+        if module.get(condition_attr) != condition_value:
+            logger.debug("Condition: %r != %r", module.get(condition_attr), condition_value)
+            return
```

Comment:
`return False`
`return` without an argument returns `None`, but this function should return a boolean.